### DISCUSSION TITLE
WM8805 I2S to SPDIF board

### DISCUSSION THREAD

**Discussion by alidaf** [sle118/squeezelite-esp32#361] [Q&A] [answered]
Does anyone have experience of these [boards](https://vi.aliexpress.com/item/1005003377281370.html?spm=a2g0o.order_list.order_list_main.29.19521802DtSKlv&gatewayAdapt=glo2vnm) and can help me get it working?

Aside from power, the inputs and connections I have made are:

`MCLK  -> 0`
`LRCK  -> 25`
`BCLK  -> 33`
`SDATA -> 32`

And the dac_config in the NVS Editor is:

`model=I2S,bck=33,ws=25,do=32,mck=0`

Unfortunately, this results in a recovery mode that is very difficult to get out of.

**Comment by alidaf** [comment]
Wut! 

I've been persisting with this and modified the autoexec1 to log debug output. I also turned off telnet and now, lo and behold, it works!
I've not changed anything else.

The only minor issue is that the bit-rate isn't being reported by the DAC in my amp so I'll investigate that a little further before closing this out.

**Reply by sle118** [reply, marked as answer]
Being thrown back to recovery is an indication that something was doing with your configuration. Ideally this requires full logs and full NVS config to be reported if you want help. There could have been something wrong with your autoexec1 configuration, as the other entries you mentioned shouldn't make a difference, but we will never know 😉

**Reply by alidaf** [reply]
@sle118 I wasn't going to take that standing up so I turned telnet back on, and.... well, I think you know the answer to that :blush:! Still, after wiping the egg off my face, it's another potential hardware option for the project. Is there a definitive list anywhere of what has been tried and works, with relevant configs to add this to?

I am curious about the bit-rate not being reported though. Is this likely to be due to the implementation of the I2S to SPDIF board, or is the I2S stream lacking some sync signal. The sound seems fine though, through both the optical and the RCA connectors. The DAC reports the bit-rate when using the digital out (SPDIF and RCA) from a CD player and from a Squeezebox Classic, as well as USB audio from a RPi running PcP.
